storage/postgres/metadata.go

```diff
@@ -68,40 +70,124 @@
 	return meta, nil
 }
 
-func (s *PGStore) SaveMeta(ty string, id string, m core.Metadata) error {
+func (s *PGStore) SaveMeta(id, timestamp, targetType, targetID, key, value string) error {
 	tx, _ := s.Conn().Begin(context.TODO())
 
-	for key, value := range m {
-		ib := sqlbuilder.NewInsertBuilder()
-		ib.InsertInto(s.table("metadata"))
-		ib.Cols(
-			"meta_target_type",
-			"meta_target_id",
-			"meta_key",
-			"meta_value",
-		)
-		ib.Values(ty, id, key, string(value))
+	ib := sqlbuilder.NewInsertBuilder()
+	ib.InsertInto(s.table("metadata"))
+	ib.Cols(
+		"meta_id",
+		"meta_target_type",
+		"meta_target_id",
+		"meta_key",
+		"meta_value",
+		"timestamp",
+	)
+	ib.Values(id, targetType, targetID, key, string(value), timestamp)
 
-		sqlq, args := ib.BuildWithFlavor(sqlbuilder.PostgreSQL)
+	sqlq, args := ib.BuildWithFlavor(sqlbuilder.PostgreSQL)
 
-		_, err := tx.Exec(
-			context.TODO(),
-			sqlq,
-			args...,
-		)
+	_, err := tx.Exec(
+		context.TODO(),
+		sqlq,
+		args...,
+	)
 
-		if err != nil {
-			tx.Rollback(context.TODO())
+	if err != nil {
+		tx.Rollback(context.TODO())
 
-			return err
-		}
+		return err
 	}
 
-	err := tx.Commit(context.TODO())
+	err = tx.Commit(context.TODO())
 
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (s *PGStore) FindMeta(q query.Query) (query.Cursor, error) {
+	q.Limit = int(math.Max(-1, math.Min(float64(q.Limit), 100))) + 1
+
+	c := query.Cursor{}
+
+	in := sqlbuilder.NewSelectBuilder()
+	in.Select(
+		"meta_id",
+		"meta_target_type",
+		"meta_target_id",
+		"meta_key",
+		"meta_value",
+	)
+	in.From("metadata")
+	in.OrderBy("meta_id desc")
+	in.Limit(q.Limit)
+
+	sqlq, args := in.BuildWithFlavor(sqlbuilder.PostgreSQL)
+	if viper.GetBool("debug") {
+		fmt.Println(sqlq, args)
+	}
+
+	rows, err := s.Conn().Query(
+		context.TODO(),
+		sqlq,
+		args...,
+	)
+
+	if err != nil {
+		return c, err
+	}
+
+	foundMetadata := map[int64]core.Metadata{}
```

Comment:
This is functionally identical. I'm going to leave this unchanged if this is just a style preference because this style matches all but one of the other empty map initializations in the repo